core/src/packet_deserializer.rs

```diff
@@ -130,10 +144,34 @@ impl PacketDeserializer {
     fn deserialize_packets<'a>(
         packet_batch: &'a PacketBatch,
         packet_indexes: &'a [usize],
+        tx_account_lock_limit: usize,
     ) -> impl Iterator<Item = ImmutableDeserializedPacket> + 'a {
-        packet_indexes.iter().filter_map(move |packet_index| {
-            ImmutableDeserializedPacket::new(packet_batch[*packet_index].clone(), None).ok()
-        })
+        packet_indexes
+            .iter()
+            .filter_map(move |packet_index| {
+                ImmutableDeserializedPacket::new(packet_batch[*packet_index].clone(), None).ok()
+            })
+            .filter(move |packet| Self::check_account_locks_limit(packet, tx_account_lock_limit))
+    }
+
+    fn check_account_locks_limit(
+        packet: &ImmutableDeserializedPacket,
+        tx_account_lock_limit: usize,
+    ) -> bool {
+        let message = &packet.transaction().get_message().message;
+        let num_static_accounts = message.static_account_keys().len();
+        let num_looked_up_accounts: usize = message
+            .address_table_lookups()
```

Comment:
We're now looping over these an extra time, should be small; not particularly concerned this will degrade performance